tests/parser/features/decorators/test_private.py

```diff
@@ -12,4 +12,387 @@ def returnten() -> int128:
+    if a + b + c + d == 1338:
+        self.greeting = _greeting
+
+@public
+def construct(greet: bytes[20]) -> bytes[40]:
+    return concat(self.greeting, greet)
+
+@public
+def iprefer(_greeting: bytes[20]):
+    a: uint256 = 112
+    b: uint256 = 211
+    self.set_greeting(_greeting)
+    assert a == 112
+    assert b == 211
+
+@public
+def hithere(name: bytes[20]) -> bytes[40]:
+    d: bytes[40] = self.construct(name)
+    return d
+    """
+
+    c = get_contract_with_gas_estimation(private_test_code)
+    assert c.greeting() == b"Hello "
+    assert c.hithere(b"Bob") == b"Hello Bob"
+    c.iprefer(b'Hi there, ', transact={})
+    assert c.hithere(b"Alice") == b"Hi there, Alice"
+
+
+def test_private_default_parameter(get_contract_with_gas_estimation):
+    private_test_code = """
+@private
+def addition(a: uint256, b: uint256 = 1) -> uint256:
+    return a + b
+
+
+@public
+def add_one(a: uint256) -> uint256:
+    return self.addition(a)
+
+@public
+def added(a: uint256, b: uint256) -> uint256:
+    c: int128 = 333
+    d: uint256 = self.addition(a, b)
+    assert c == 333
+
+    return d
+    """
+
+    c = get_contract_with_gas_estimation(private_test_code)
+
+    assert c.add_one(20) == 21
+    assert c.added(10, 20) == 30
+
+
+def test_private_return_tuple(get_contract_with_gas_estimation):
+    code = """
+@private
+# @public
+def _test(a: int128) -> (int128, int128):
+    return a + 2, 2
+
+
+@public
+def test(a: int128) -> (int128, int128):
+    return self._test(a)
+    """
+
+    c = get_contract_with_gas_estimation(code)
+
+    assert c.test(11) == [13, 2]
+
+
+def test_private_return_bytes(get_contract_with_gas_estimation):
+    code = """
+a_message: bytes[50]
+
+@private
+def _test() -> (bytes[100]):
+    b: bytes[50] = "hello                   1           2"
+    return b
+
+@private
+def _test_b(a: bytes[100]) -> (bytes[100]):
+    b: bytes[50] = "hello there"
+    if len(a) > 1:
+        return a
+    else:
+        return b
+
+@private
+def get_msg() -> (bytes[100]):
+    return self.a_message
+
+@public
+def test() -> (bytes[100]):
+    d: bytes[100]
+    d = self._test()
+    return d
+
+@public
+def test2() -> (bytes[100]):
+    d: bytes[100] = 'xyzxyzxyzxyz'
+    return self._test()
+
+@public
+def test3(a: bytes[50]) -> (bytes[100]):
+    d: bytes[100] = 'xyzxyzxyzxyz'
+    return self._test_b(a)
+
+@public
+def set(a: bytes[50]):
+    self.a_message = a
+
+@public
+def test4() -> (bytes[100]):
+    d: bytes[100] = 'xyzxyzxyzxyz'
+    return self.get_msg()
+    """
+
+    c = get_contract_with_gas_estimation(code)
+
+    assert c.test() == b"hello                   1           2"
+    assert c.test2() == b"hello                   1           2"
+    assert c.test3(b"alice") == b"alice"
+    c.set(b"hello daar", transact={})
+    assert c.test4() == b"hello daar"
+
+
+def test_private_bytes_as_args(get_contract_with_gas_estimation):
+    code = """
+@private
+def _test(a: bytes[40]) -> (bytes[100]):
```

Comment:
I think the style guide should *suggest* that `_name` be the convention for private functions.